### ISSUE TITLE
How to use it

### ISSUE THREAD

**Issue by GeoffreyMba** [SK83RJOSH/rust-jc2-tools#1]
I downloaded all the files except arc and when I click on jc2_tools.exe it opens two windows then later closes them. How do it use these tools?

**Comment by SK83RJOSH** [comment]
Currently, the tool only allow you to browse archives and view a small subset of models in the game - so with that said, it sounds like you don't have vulkan drivers or similar?

I can't really triage without additional information, like a video or screenshot of the console output when you run the tool.

One of the windows is a console, which will give me useful information:
![image](https://github.com/user-attachments/assets/74138da2-868a-43e8-8bb6-fa0f9ebb1ed2)

--

If you do get it working though, you'll want to mount content like so:
![image](https://github.com/user-attachments/assets/efc6449a-e5ac-4c64-9984-ce0b65ded717)
![image](https://github.com/user-attachments/assets/6dea0c5b-4beb-4ebe-906f-37c3e51ee9b5)

Once complete, you can right click files in the file list on the left to load them:
![image](https://github.com/user-attachments/assets/70bf900a-a92f-4a4f-a637-caa8f9592a36)
![image](https://github.com/user-attachments/assets/aac97261-d7cd-4ff1-8e05-2d6b2b34c31d)
![image](https://github.com/user-attachments/assets/2347d90e-e7dc-4008-a14d-846e7337c03c)

**Comment by GeoffreyMba** [comment]
Yeah that black console shows up for a second then just disappears. I'll
try to get a screenshot. Also the rbm to gltf converter has some issues.
When I try to open the output gltf into cad assistant or Windows 3D viewer,
it says it can't open the file even tho gltf files are supported by both
programs. Tho I used gibbed tools to small unpack the .ee archive then I
took the .rbm e.g Jack-lod-1a.rbm and placed it on the converter.exe. A
black console shows up then disappears then I see the output file. If you
can sir can you explain to me the entire process of getting for example "
ARVE.V073_DLC_Jet.ee" from the dlc tab archive all the way to at least .obj
3d format? Thank you sir

**Comment by SK83RJOSH** [comment]
I have only tested glTF with Blender, and a few online viewers. By the
sounds of it, those tools perhaps don't support binary glTF files? Or my
file might have some validation errors – which is likely, Blender is fairly
forgiving.

Do note the glTF files do not include texture data, and you'll still need
to use gibbeds tools to dump those.

—

As far as the app goes, you're not missing much besides a better file
explorer. You can steal my file lists and add them to gibbeds tools though,
and that gets you pretty much the same thing.

**Comment by GeoffreyMba** [comment]
Here are the screenshots you asked for

**Comment by SK83RJOSH** [comment]
Looks like GitHub doesn't support attachments via email – any chance you could attach them here? :)

**Comment by GeoffreyMba** [comment]
![Screenshot (5)](https://github.com/user-attachments/assets/caa6bd14-6317-40c0-873f-8169faa00fb6)
![Screenshot (4)](https://github.com/user-attachments/assets/e4590df2-c6ef-4818-a47b-bb1b5be7ad0d)

**Comment by GeoffreyMba** [comment]
I just resent them.

**Comment by SK83RJOSH** [comment]
Hmm, yeah looks like it's graphics issue indeed – I it must be related to integrated graphics but I'd need to see an error to know for sure – you might be able to start the tool from CMD/PowerShell and redirect it's console output that way.

In the meantime I did a quick search for related issues in `bevy` and didn't see much. Though I'm preparing an upgrade to the latest version, which will hopefully solve this as its had a lot of graphics improvements / uses a newer version of WGPU – but I'm currently blocked while I wait for plugins to update.

**Comment by GeoffreyMba** [comment]
Pls tell me how to use power shell to start the app

**Comment by SK83RJOSH** [comment]
`shift + right click` in the folder, and `Open PowerShell Here`.

Then write `jc2-tools.exe` and press enter

**Comment by GeoffreyMba** [comment]
If there are errors, will they show me?

**Comment by SK83RJOSH** [comment]
I haven't tested it, but you might get some additional info – but frankly I suggest you try again in a week or so when I have a chance to update to the latest version of bevy. :)

**Comment by SK83RJOSH** [comment]
I've updated to bevy 0.14, go ahead and give the latest release a try and see if that resolves the issue.

To run jc2_tools in powershell type:
`./jc2_tools.exe`

Make sure you're in the correct directory. It put the full error log into that window and you can send it along to me. 👍

**Comment by GeoffreyMba** [comment]
Okay thanks. JUst to clarify, I should try if the app works straight but if
it doesn't, I should try using power shell and get you the error log right?

**Comment by SK83RJOSH** [comment]
Yeap, hopefully the upgrade fixed the issue, but worst case scenario we can
still try to get some additional info.

**Comment by GeoffreyMba** [comment]
The app opens but I cannot explore my archives. It just goes the scroll bar
is empty and it doesn't respond when I try to drag a file on it. I used
power shell and I got a report but when I charge my laptop, I'll send it to
you.

**Comment by SK83RJOSH** [comment]
Do make you mounted content – you have to do so from the options menu and select the folder containing `justcause2.exe` – until you've done that it won't show the file tree. Similarly, you must use the file menu to open rbm files directly. The app does not support drag and drop yet.

**Comment by GeoffreyMba** [comment]
Wait I'll send you some screenshots.

**Comment by GeoffreyMba** [comment]
![Screenshot (23)](https://github.com/user-attachments/assets/a0ea5003-021f-46d0-bb6a-69e8ad9ea6db)
![Screenshot (22)](https://github.com/user-attachments/assets/0da07f15-5116-4da4-8531-f3d2ac44e753)
![Screenshot (21)](https://github.com/user-attachments/assets/ad6d87ca-16e4-419c-93cb-5e8560403633)

**Comment by SK83RJOSH** [comment]
This seems to be a wgpu-hal/bevy bug:
https://github.com/bevyengine/bevy/issues/13115
https://github.com/bevyengine/bevy/issues/14213

Though to be honest, I think this is more of an issue with your processors graphics drivers.

Try doing the following in PowerShell: 
```
$env:WGPU_BACKEND="DX12"
./jc2_tools.exe
```

You can also try: 
```
$env:WGPU_BACKEND="GL"
./jc2_tools.exe
```

Not a lot a can do for you though, I don't really want to force a particular WGPU backend on end users.

**Comment by GeoffreyMba** [comment]
Thanks for trying!
